docs/core/howto/tap.rst

```diff
@@ -310,6 +310,107 @@ For a full list of cred plugins supported, see :api:`twisted.plugins <twisted.pl
 
 
 
+Deploy your application using python packages
+---------------------------------------------
+
+
+
+To deploy your application one possibility is to wrap it up in a python package.
+For this you need to write a special file ``setup.py``, which contains metadata
+of the package. You would have to extend the layout of your files like this:
+
+
+
+
+- **MyProject** - Top level directory
+
+
+- ``setup.py`` - Description file for the package
+
+
+  - **myproject** - Python package
+
+
+    - **__init__.py**
+
+
+  - **twisted**
+
+
+    - **plugins**
+
+      - ``myplugin.py`` - Plugin file
+
+
+
+
+
+.. code-block:: python
+
+
+    from distutils.core import setup
+
+    setup(
+        name='MyApplication',
+        version='0.1dev',
+        packages=['myproject', 'twisted.plugins'],
+        license='Creative Commons Attribution-Noncommercial-Share Alike license',
+        long_description="Insert your long description here",
+        install_requires=[
+            'twisted',
+            ],
+    )
+
+
+
+
+
+To create the python package from the directory the standard setup tools
+can be used:
+
+
+
+
+.. code-block:: console
+
+
+    python setup.py sdist
```

Comment:
wheels are the best, right?  Any reason to recommend sdist instead of wheel here?

Reply:
My understanding is that the best practice is to consider the `sdist` the "root" thing, which you should always upload, and wheels are derivative artifacts of that.  The differences are mostly only relevant when it comes time to start building native extensions, of course.

Reply:
(That said I *totally* just upload wheels to PyPI for a bunch of my smaller projects and direct people to git repos if they want to build from source.)